api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,6 +87,11 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// ClientCertRef defines the reference to a Kubernetes Secret that contains
+	// the client certificate and private key for Envoy to use when connecting to
+	// backend services and external services, such as ExtAuth, ALS, OpenTelemetry, etc.
+	ClientCertRef *gwapiv1.SecretObjectReference `json:"clientCertRef,omitempty"`
```

Comment:
How can we specify which backend service enforces client validation and which doesn't require it?

Reply:
AFAIK, if the the TLS `server_hello` doesn't include a `certificate_request`, then the client [does not send the certificate](https://datatracker.ietf.org/doc/html/rfc5246#section-7.4.6). 

I can still think of some scenarios where sending a "technical" client cert when asked would be a problem. Some servers support multiple means of auth (client-cert, token, ... ), e.g. the K8s API server: https://kubernetes.io/docs/reference/access-authn-authz/authentication/. When proxying such a server, EG may be requested to present a client cert, and will present it. The server may try to authenticate the end-user based on the EG client cert, when the end-user actually intended for auth to be based on some L7 token. This will break such an integration.